File: woodroid/src/main/java/me/gilo/woodroid/data/api/CartAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.LineItem;
import me.gilo.woodroid.models.filters.CartFilter;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;
import java.util.Map;

public interface CartAPI {

    @Headers("Content-Type: application/json")
    @POST("clear")
    Call<String> clear();

    @GET("count-items")
    Call<Integer> count();

    @GET("cart")
    Call<Map<String, LineItem>> list();

    @Headers("Content-Type: application/json")
    @POST("cart/add")
    Call<Map<String, LineItem>> addToCart(@Body LineItem body);

    @DELETE("cart/cart-item")
    Call<String> delete(@Body CartFilter body);

    @GET("cart/cart-item")
    Call<String> restore(@Body CartFilter body);

    @Headers("Content-Type: application/json")
    @POST("cart/cart-item")
    Call<String> update(@Body CartFilter body);


}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/CartAPI.kt
package me.gilo.woodroid.data.api


import retrofit2.Call
import me.gilo.woodroid.models.LineItem
import me.gilo.woodroid.models.filters.CartFilter
import retrofit2.http.*

interface CartAPI {

    @Headers("Content-Type: application/json")
    @POST("clear")
    fun clear(): Call<String>

    @GET("count-items")
    fun count(): Call<Int>

    @GET("cart")
    fun list(): Call<Map<String, LineItem>>

    @Headers("Content-Type: application/json")
    @POST("cart/add")
    fun addToCart(@Body body: LineItem): Call<Map<String, LineItem>>

    @DELETE("cart/cart-item")
    fun delete(@Body body: CartFilter): Call<String>

    @GET("cart/cart-item")
    fun restore(@Body body: CartFilter): Call<String>

    @Headers("Content-Type: application/json")
    @POST("cart/cart-item")
    fun update(@Body body: CartFilter): Call<String>


}